form.py

```diff
@@ -555,53 +622,78 @@ def command_line_convert(origin, target):
+
+    if mode is None:
+        mode = get_convert_mode(origin)
+
+    print_(f"Getting form HTML source [{mode}]: {origin}")
+    text = get_html_from_convert(origin, mode)
+
+    print_("Converting form...")
     soup = BeautifulSoup(text, "html.parser")
     info = info_using_soup(soup) | info_using_json(form_json_data(soup))
 
     # Write the info to the config file
-    print("Writing to config file...")
+    print_(f"Writing to config file: {target}")
     with open(target, mode="w") as file:
         for line in config_lines_from_info(info):
             file.write(line + "\n")
 
-    print(f"Form converted and written to file: {target}")
+    print_(f"Form converted and written to file: {target}")
+
+# Pass in sys.argv[1:]. Returns whether the program was run using a double
+# click of drag and dropped on.
+def is_simple_run(argv):
+    if len(argv) == 0:  # Double click
+        return True
+    if len(argv) == 1:  # Drag and dropped file is argument
+        if argv[0] not in "--help -h process p convert c".split():
+            return True
+    return False
+
+# Pass in sys.argv[1:]. Assume is_simple_run(argv) is True. Returns converted
+# arguments that can be passed into better.parse_args.
+def convert_simple_argv(argv):
+    if not argv:  # Double click
+        return ["process", "config.txt"]
+    else:  # Drag and dropped file is argument
+        return [get_target_command(argv[0]), argv[0]]
 
 def main(args):
-    # If neither --process or --convert was specified
-    if not args.process and not args.convert:
-        if get_target_action(args.target) == "process":
-            args.process = True
-        else:
-            args.process = False
-            args.convert, args.target = args.target, "config.txt"
-
-    if args.process:
-        command_line_process(args.target)
-    if args.convert:
-        command_line_convert(args.convert, args.target)
+    if args.command in "process p".split():
+        return process(args.target, command_line=True)
+    if args.command in "convert c".split():
+        return convert(args.origin, args.target, args.mode, command_line=True)
+    raise ValueError(f"Unknown command: {args.command}")
 
 if __name__ == "__main__":
-    # Not wrapped by try-finally as the user is likely running this from the
-    # command line.
-    args = parser.parse_args()
-
+    argv = sys.argv[1:]
+    simple_run = is_simple_run(argv)
     try:
+        if simple_run:
+            argv = convert_simple_argv(argv)
+        args = better.parse_args(argv)
         main(args)
     except Exception:  # This won't catch Ctrl+C or sys.exit
+        if not simple_run:
+            raise
+        # Printed and replaced with sys.exit as the user is likely running this
+        # using a double click or a drag and drop.
         traceback.print_exc()
         sys.exit(4)
     finally:
-        input("Press enter to close the program...")
+        if simple_run:
+            with suppress(BaseException):
```

Comment:
*BaseException* is a bit overkill... Maybe just *KeyboardInterrupt* (and *EOFError* if you really need it).

Reply:
Replaced with *KeyboardInturrupt* in 7e5fd6f1b82d5d7af47f8b53d95df47844451ce3.